packages/sources/coinpaprika/src/endpoint/crypto.ts

```diff
@@ -76,95 +78,42 @@ export const inputParameters: InputParameters = {
-
-  requestedData.forEach(({ coinid, symbol }) => {
-    const coin = getCoin(responseData, symbol, coinid)
-    if (!coin) {
-      throw new Error(`unable to find coin: ${coinid || symbol}`)
-    }
-
-    for (const quote in coin.quotes) {
-      payload.push([
-        {
-          ...request,
-          data: {
-            ...request.data,
-            base: coin.symbol.toUpperCase(),
-            quote: quote.toUpperCase(),
-          },
-        },
-        Requester.validateResultNumber(coin, ['quotes', quote, resultPath]),
-      ])
-    }
-  })
-
-  // We'll reset the response data to not output the entire CP coins list
-  const result = Requester.withResult({ ...response, data: {} }, undefined, payload)
-  return Requester.success(jobRunID, result, true, batchablePropertyPath)
-}
-
-export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
-  const validator = new Validator(request, inputParameters, {}, { overrides })
+export const execute: ExecuteWithConfig<Config> = async (request, context, config) => {
+  const validator = new Validator(request, inputParameters)
 
   const jobRunID = validator.validated.id
-  const symbol = validator.overrideSymbol(AdapterName)
+  const base: string | string[] = validator.validated.data.base
   const requestedQuotes = validator.validated.data.quote
   const coinid = validator.validated.data.coinid as string | undefined
 
-  const url = 'v1/tickers'
   const resultPath = validator.validated.data.resultPath || endpointResultPaths.crypto
 
-  let quotes: string
-  if (Array.isArray(requestedQuotes)) {
-    quotes = requestedQuotes.map((quote) => quote.toUpperCase()).join(',')
-  } else {
-    quotes = requestedQuotes.toUpperCase()
-  }
+  const quotes = Array.isArray(requestedQuotes)
+    ? requestedQuotes.map((quote) => quote.toUpperCase()).join(',')
+    : requestedQuotes.toUpperCase()
 
-  const params = { quotes }
+  const requestedCoins = await getConvertedCoins(jobRunID, base, request.data.overrides, context)
   const options = {
     ...config.api,
-    url,
-    params,
+    url: 'v1/tickers',
+    params: { quotes },
   }
 
-  if (Array.isArray(symbol)) {
-    const requestedData: RequestedData[] = []
-
-    for (let i = 0; i < symbol.length; i++) {
-      if (symbol[i] !== validator.validated.data.base[i]) {
-        requestedData.push({ coinid: symbol[i] })
-      } else {
-        requestedData.push({ symbol: symbol[i] })
-      }
-    }
-
+  if (Array.isArray(base)) {
     const response = await Requester.request<ResponseSchema[]>(options)
-    return handleBatchedRequest(jobRunID, request, response, requestedData, resultPath)
+    return handleBatchedRequest(jobRunID, request, response, requestedCoins, resultPath)
   }
 
-  // If coinid was provided or base was overridden, that symbol will be fetched
-  const coin = coinid || (symbol !== validator.validated.data.base && symbol ? symbol : undefined)
-
+  options.url += '/' + requestedCoins[base]
```

Comment:
Why is the functionality of this endpoint being changed?
This effectively is making it the `crypto-single` endpoint.

If the reason is an optimization to avoid large payloads:
1. I'd prefer to use an endpoint override (e.g. [in Coinapi](https://github.com/smartcontractkit/external-adapters-js/blob/develop/packages/sources/coinapi/src/endpoint/crypto.ts#L7))
2. Overriding the endpoint has a flaw - we first make a single request with the full response as that is what the batch warmer uses and we want to ensure the single request can be served from the large response. What if `/tickers/ETH` works, but `ETH` isn't contained in `/tickers`? Now we will have a single feed making it into the batch wamer that then repeatedly kills it because it errors on `ETH`

Reply:
Gotcha.  I will change this.

Reply:
The most straightforward way to make this work is, if a single symbol is sent to the crypto endpoint as a string, I will wrap that symbol in an array and remove the logic for handling single symbols.